crates/assists/src/handlers/generate_setter.rs

```diff
@@ -0,0 +1,157 @@
+use ast::Adt;
+use stdx::{format_to, to_lower_snake_case};
+use syntax::ast::VisibilityOwner;
+use syntax::ast::{self, AstNode, NameOwner};
+
+use crate::{
+    utils::{find_impl_block, find_struct_impl, generate_impl_text},
+    AssistContext, AssistId, AssistKind, Assists,
+};
+
+// Assist: generate_setter
+//
+// Generate a setter method.
+//
+// ```
+// struct Person {
+//     nam$0e: String,
+// }
+// ```
+// ->
+// ```
+// struct Person {
+//     name: String,$0
+// }
+//
+// impl Person {
+//     /// Set the person's name.
+//     fn set_name(&mut self, name: String) {
+//         self.name = name;
+//     }
+// }
+// ```
+pub(crate) fn generate_setter(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let strukt = ctx.find_node_at_offset::<ast::Struct>()?;
+    let field = ctx.find_node_at_offset::<ast::RecordField>()?;
+
+    let field_name = field.name()?;
+    let field_ty = field.ty()?;
+
+    let fn_name = to_lower_snake_case(&field_name.to_string());
+
+    // Return early if we've found an existing fn
+    let impl_def =
+        find_struct_impl(&ctx, &Adt::Struct(strukt.clone()), format!("set_{}", fn_name).as_str())?;
+
+    let target = field.syntax().text_range();
+    acc.add(
+        AssistId("generate_setter", AssistKind::Generate),
+        "Generate a setter method",
+        target,
+        |builder| {
+            let mut buf = String::with_capacity(512);
+
+            if impl_def.is_some() {
+                buf.push('\n');
+            }
+
+            let vis = strukt.visibility().map_or(String::new(), |v| format!("{} ", v));
+            format_to!(
+                buf,
+                "    /// Set the person's name.
```

Comment:
```suggestion
                "    /// Set the {}'s {}.
```
You forgot to template here.